.github/workflows/llvm-ci-pre-land.yml

```
@@ -0,0 +1,134 @@
+# CI jobs to be run upon the code lands to the main branch or GitHub Action test branches.
+
+name: llvm-ci-pre-land
+
+on:
+  pull_request:
+    branches: [main, llvm-sys, gha-test-*]
```

not sure how many branches we will have for developing. afaik we only have `solana` branch atm.

```suggestion
    branches: [solana]
```